api/v1alpha1/etcd_types.go

```diff
@@ -249,29 +225,40 @@ type LastOperation struct {
 	Type LastOperationType `json:"type,omitempty"`
 }
 
-// Status defines the observed state of Etcd
-type Status struct {
-	Etcd            CrossVersionObjectReference `json:"etcd,omitempty"`
-	Conditions      []Condition                 `json:"conditions,omitempty"`
-	CurrentReplicas int32                       `json:"currentReplicas,omitempty"`
-	Endpoints       []corev1.Endpoints          `json:"endpoints,omitempty"`
-	LastError       string                      `json:"lastError,omitempty"`
-	Replicas        int32                       `json:"replicas,omitempty"`
-	ReadyReplicas   int32                       `json:"readyReplicas,omitempty"`
-	Ready           bool                        `json:"ready,omitempty"`
-	UpdatedReplicas int32                       `json:"updatedReplicas,omitempty"`
+// EtcdStatus defines the observed state of Etcd
+type EtcdStatus struct {
+	// +optional
+	Etcd CrossVersionObjectReference `json:"etcd,omitempty"`
+	// +optional
+	Conditions []Condition `json:"conditions,omitempty"`
+	// +optional
+	CurrentReplicas int32 `json:"currentReplicas,omitempty"`
+	// +optional
+	Endpoints []corev1.Endpoints `json:"endpoints,omitempty"`
+	// +optional
+	ServiceName *string `json:"serviceName,omitempty"`
+	// +optional
+	LastError *string `json:"lastError,omitempty"`
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+	// +optional
+	ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+	// +optional
+	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
```

Comment:
I will insist on the `LabelSelector`. Technically, it might be optional in `Scale` sub-resource. But neither HPA nor VPA (nor HVPA, for that matter) would work without it. HPA not working may not be important. But VPA and HVPA not working would be a non-starter.

Reply:
ok. Will add it.

Reply:
👍 Thanks